File: bot/src/main/java/net/slimediamond/atom/discord/commands/minecraftonline/VoteCommand.java
package net.slimediamond.atom.discord.commands.minecraftonline;

import net.dv8tion.jda.api.EmbedBuilder;
import net.slimediamond.atom.command.discord.DiscordCommandContext;
import net.slimediamond.atom.command.discord.DiscordCommandExecutor;
import net.slimediamond.atom.reference.EmbedReference;

import java.awt.*;
import java.util.List;
import java.util.stream.Collectors;

public class VoteCommand implements DiscordCommandExecutor {

    private static final List<VotingSite> SITES = List.of(
            new VotingSite("Minecraft Server List", "https://minecraft-server-list.com/server/267113/vote/"),
            new VotingSite("Planet Minecraft", "https://www.planetminecraft.com/server/minecraft-online-3375846/vote/")
    );

    @Override
    public void execute(DiscordCommandContext context) throws Exception {
        System.out.println("Command invoked");
        context.replyEmbeds(new EmbedBuilder()
                .setColor(Color.GREEN)
                .setAuthor("MinecraftOnline - Voting sites", null, "https://i.slimediamond.net/64px-Gold_Ingot_JE1-4064757708.png")
                .setDescription(VoteCommand.SITES.stream()
                        .map(site -> " * [" + site.title + "](" + site.url + ")")
                        .collect(Collectors.joining("\n")))
                .setFooter(EmbedReference.mcoFooter, EmbedReference.mcoIcon)
                .build());
    }

    private record VotingSite(String title, String url) {

    }

}

File: bot/src/main/kotlin/net/slimediamond/atom/commands/minecraftonline/VoteCommand.kt
package net.slimediamond.atom.commands.minecraftonline

import net.slimediamond.atom.api.command.CommandNode
import net.slimediamond.atom.api.command.CommandNodeContext
import net.slimediamond.atom.api.command.CommandResult
import net.slimediamond.atom.api.command.platforms.discord.DiscordCommandNodeContext
import net.slimediamond.atom.api.discord.embed.description
import net.slimediamond.atom.api.messaging.RichText
import net.slimediamond.atom.api.messaging.richText
import net.slimediamond.atom.utils.Embeds

class VoteCommand : CommandNode("Show voting sites for MinecraftOnline", "vote") {

    // TODO: Configurate this (:
    companion object {
        private val VOTING_SITES = listOf(
            Site("Minecraft Server List", "https://minecraft-server-list.com/server/267113/vote/"),
            Site("Planet Minecraft", "https://www.planetminecraft.com/server/minecraft-online-3375846/vote/")
        )
    }

    override suspend fun execute(context: CommandNodeContext): CommandResult {
        if (context is DiscordCommandNodeContext) {
            context.sendEmbed {
                title = "MinecraftOnline voting sites"
                description {
                    VOTING_SITES.forEach { site ->
                        append(richText("* [${site.name}](${site.url})\n"))
                    }
                }
                footer = Embeds.MCO_FOOTER
            }
        } else {
            context.sendMessage {
                append(richText("Voting sites for MinecraftOnline: "))
                append(RichText.join(richText(", "), VOTING_SITES.stream().map { richText(it.url) }.toList()))
            }
        }
        return CommandResult.success
    }

    data class Site(val name: String, val url: String)

}